AmericanMedicalAssociation/ama-style-guide-2
# EWL-10995: Update page grouper design and functionality on mobile

# Pre-PR Checklist
If you have access to approved developer tools such as AMA co-pilot, ask it the following and test any suggestions before posting the PR.


## JIRA Ticket(s)
- [EWL-10995: Mobile | Page Grouper redesign](https://ama-it.atlassian.net/browse/EWL-10995)


## Description:
- Adjust styling and functionality of Article Page Grouper blocks on mobile.
- Add drawer functionality on mobile
- Add sticky functionality to page grouper on mobile
- Add script to hide locker navigation on scroll when page grouper overlaps (this is a design constraint)


## To Test:
- checkout ama-d8 pr: https://github.com/AmericanMedicalAssociation/ama-d8/pull/4998
- CLEAR CACHES
- Use these zeplins for all following checks: https://app.zeplin.io/project/66c7d90db37684fc8708e91d
- add/go to article with page grouper
- on mobile confirm the page grouper collapses into a drawer
- confirm on click of title drawer opens, on click again the drawer closes
- open drawer and confirm you can not scroll the rest of the page
- when drawer opened confirm you can click anywhere outside of page grouper to close drawer/menu
- confirm on scroll the drawer becomes sticky
- confirm on scroll once page grouper overlaps the subscribe menu (the three dots in the top right corner) the dots disappear (note there is a timer on this so it wont be immediate, this is to lessen the impact on performance)
- edit current page grouper block to have 6+ links
- re-publish article and on mobile confirm drawer opens and can be scrolled within to see other links
- **NOTE**: on mac scroll bars are hidden by default until scrolled, this is a system setting we can not update, let me know and I can show how to turn this off to see the scrolling capability by default
- confirm all focus and hover states work as shown in the zeplins
- confirm desktop styling has not changed

Edge Case:
- make title of page grouper a large number of characters
- confirm on mobile the text cuts off and shows elipses


## Visual Regressions
Ensure any available regression testing has been run.


## Relevant Screenshots/GIFs
Use something like [Skitch](https://evernote.com/skitch/) or [GIPHY Capture](https://giphy.com/apps/giphycapture) to capture images/gifs to demonstrate behaviors.


## Remaining Tasks



## Additional Notes
Anything more to add?

---

[Guidelines for Contribution](CONTRIBUTING.md)
[SG2 Standards](ama-style-guide-2/docs/standards.md)
